src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaBaseClassTests.cs

```diff
@@ -0,0 +1,225 @@
+        }
+
+        [Theory]
+        [MemberData(nameof(AlgorithmsData))]
+        public static void CallsSignDataCore(SlhDsaAlgorithm algorithm)
+        {
+            SlhDsaTestImplementation slhDsa = SlhDsaTestImplementation.CreateOverriddenMethodsFail(algorithm);
+
+            int signatureSize = algorithm.SignatureSizeInBytes;
+            byte[] signature = new byte[signatureSize + 2];
+            byte[] testData = [2];
+            byte[] testContext = [3];
+
+            slhDsa.SignDataCoreHook = (ReadOnlySpan<byte> data, ReadOnlySpan<byte> context, Span<byte> destination) =>
+            {
+                Assert.Equal(testData, data);
+                Assert.Equal(testContext, context);
+
+                Assert.Equal(destination.Length, signatureSize);
+                destination.Fill(1);
+            };
+
+            Array.Fill(signature, (byte)42);
+            slhDsa.SignData(testData, signature.AsSpan(1, signatureSize), testContext);
+
+            Assert.Equal(42, signature[0]);
+            Assert.Equal(42, signature[^1]);
+            Assert.All(signature.Skip(1).SkipLast(1), b => Assert.Equal(1, b));
+
+            Array.Fill(signature, (byte)42);
+            slhDsa.SignData(testData, signature.AsSpan(1, signatureSize), testContext); // Extra byte should be ignored
+
+            Assert.Equal(42, signature[0]);
+            Assert.Equal(42, signature[^1]);
+            Assert.All(signature.Skip(1).SkipLast(1), b => Assert.Equal(1, b));
+        }
+
+        [Theory]
+        [MemberData(nameof(AlgorithmsData))]
+        public static void CallsVerifyDataCore(SlhDsaAlgorithm algorithm)
+        {
+            SlhDsaTestImplementation slhDsa = SlhDsaTestImplementation.CreateOverriddenMethodsFail(algorithm);
+
+            int signatureSize = algorithm.SignatureSizeInBytes;
+            byte[] testSignature = new byte[signatureSize + 1];
+            byte[] testData = [2];
+            byte[] testContext = [3];
+            bool returnValue = false;
+
+            Array.Fill(testSignature, (byte)42);
+
+            slhDsa.VerifyDataCoreHook = (ReadOnlySpan<byte> data, ReadOnlySpan<byte> context, ReadOnlySpan<byte> signature) =>
+            {
+                Assert.Equal(testData, data);
+                Assert.Equal(testContext, context);
+                Assert.Equal(testSignature.AsSpan(0, signatureSize), signature);
+
+                return returnValue;
+            };
+
+            returnValue = true;
+            Assert.False(slhDsa.VerifyData(testData, testSignature.AsSpan(0, signatureSize - 1), testContext));
+            Assert.False(slhDsa.VerifyData(testData, testSignature.AsSpan(0, signatureSize + 1), testContext));
+
+            Assert.True(slhDsa.VerifyData(testData, testSignature.AsSpan(0, signatureSize), testContext));
+
+            returnValue = false;
+            Assert.False(slhDsa.VerifyData(testData, testSignature.AsSpan(0, signatureSize), testContext));
```

Comment:
This one, I think, could use some comments.

```suggestion
            // Since `returnValue` is true, this shows the Core method doesn't get called for the wrong sized signature.
            returnValue = true;
            Assert.False(slhDsa.VerifyData(testData, testSignature.AsSpan(0, signatureSize - 1), testContext));
            Assert.False(slhDsa.VerifyData(testData, testSignature.AsSpan(0, signatureSize + 1), testContext));

            // But does for the right one.
            Assert.True(slhDsa.VerifyData(testData, testSignature.AsSpan(0, signatureSize), testContext));

            // And just to prove that the Core method controls the answer...
            returnValue = false;
            Assert.False(slhDsa.VerifyData(testData, testSignature.AsSpan(0, signatureSize), testContext));
```